### ISSUE TITLE
seed node and replace_address

### ISSUE THREAD

**Issue by kondetibharat** [Netflix/Priam#544]
Hi, I am trying to understand how Priam handles bootstrapping of seed nodes when they loose all their data (wiped data directory) and comes back with the same IP address.

We deploy our Cassandra cluster in a cloud environment where machines (VM) are completely wiped off and brought up again with same IP address. This can happen because of VM re-assignments to different racks/pods or by resiliency service randomly wiping the data on a VM.

We dynamically set replace_address="To its own address" as it is trying to replace itself. This works fine for non-seed node as it automatically bootstraps itself and gets all the lost data back. But the problem is with seed nodes as they are not automatically bootstrapped. I was wondering how Priam handles this scenario.

Would appreciate any help on this.

**Comment by nirajampac** [comment]
I have a similar issue too.. We use ENI's to bootstrap seed nodes with the same address.. For now I changed the AMI to use the ENI address in the PriamConfiguration to make it work .. Is there a better way to do it ?

**Comment by JnMik** [comment]
> We use ENI's to bootstrap seed nodes with the same address..

@nirajampac  I am interested to know more about this procedure, my cassandra slaves are running inside docker containers, and the IPs are always dynamic. How can I detect the right IP to use from the network interfaces ? I was trying to match the network interfaces IP to down nodes from nodetool status but I realized nodetool is not running on non-seeds nodes lol !

**Comment by nirajampac** [comment]
AWS ENI's have static IP  .. I am not running on docker.. So when I boot strap  seed nodes, I added the available  ENI  and mount it 

When I bootstrap non seed ENI's , I query the ENI's that are being used  to set the seed node IP's in the cassandra config. there is an AWS white paper where the whole design is documented .


Non seed nodes should have  everything running .. so if nodetool is not running it means cassandra is not running

**Comment by JnMik** [comment]
Exactly, nodetool is not running because cassandra is not already bootstraped, because I need to define de replace_adresse before I bootstrap it, but I can't because I do not have access to nodetool xD
Running in circles.

I think I'll just try to use cqlsh to connect remotely to the master node and inspect directly the cassandra schema instead of using nodetool. I don't like the workaround, but I don't see anything else right now.

I really don't like the needed manual intervention to have a node rejoin the cluster after a failure. That "replace_address" procedure should be handled automatically by the cluster IMO.

**Comment by JnMik** [comment]
It's still a draft, but I managed to do something like this in /etc/cassandra/cassandra-env.sh: 
```
# Get current node address from cassandra.yaml
listenIpAddress=$(cat /etc/cassandra/cassandra.yaml | grep listen_address: | awk '{print $2}')
echo IP Address found in cassandra.yaml is $listenIpAddress

# Connect to master to investigate nodetool status
QUERY_RESPONSE=$(cqlsh gateway_cluster_cassandra-master _*some-user-and-password*_ -e "select peer, host_id, rpc_address from system.peers where peer='$listenIpAddress';")
echo $QUERY_RESPONSE

if [ -z $("$QUERY_RESPONSE | grep (0 rows)") ]; then
    echo Current node ip FOUND in cluster, node will bootstrap with replace_address option and then join the cluster
    JVM_OPTS="$JVM_OPTS -Dcassandra.replace_address=$listenIpAddress"
else
    echo Current node IP NOT FOUND in cluster, node will bootstrap and join normally
fi
```

A bit questionable, but working so far.. !
